sql/sp.cc

```diff
@@ -2238,6 +2241,75 @@ Sp_handler::sp_find_routine(THD *thd, const Database_qualified_name *name,
 }
 
 
+/**
+  Determine the existence of a routine.
+  Note: ONLY for unqualified routines
```

Comment:
please, don't say "unqualified", it's ambiguous and very confusing. because db.routine is a *qualified* name, and here, in sp_find_routine_quick all routine names are qualified, because you use Database_qualified_name, which included db name.

so it's very confusing when you say "ONLY for unqualified routines" but the argument is Database_qualified_name.

What you mean, routines that don't have a package name, may be you can say "ONLY for package-unqualified routines" or "ONLY for routine names without a package name"